File: solutions/24-suzad/AgeOfVillagers/AgeOfVillagers/NationFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AgeOfVillagers
{
    public class NationFactory
    {
        public INation getNation(string nation)
        {
            if(nation=="Bangladeshi Farmer")
            {
                return new BdFarmerNation();
            }
            else if(nation=="Egyptian King")
            {
                return new EgyptianKingNation();
            }
            else if(nation=="Arab Bedouin")
            {
                return new ArabBedouinNation();
            }
            else if(nation=="Inuit Hunter")
            {
                return new InuitHunterNation();
            }
            return null;
        }
    }
}

File: solutions/40-arif-JAVA/AgeOfVillagers/src/nation/NationFactory.java
package nation;

import canvas.MyCanvas;

public class NationFactory {

    public INation getNation(String nationType) {

        if (nationType.equals( "Bangladeshi Farmer"))
            return new BDFarmers();
        else if (nationType.equals("Arab Bedouin"))
            return new ArabBedouins();
        else if (nationType.equals( "Egyptian King"))
            return new EgyptianKings();
        else if (nationType.equals( "Inuit Hunter"))
            return new InuitHunters();
        else {
            NoNationSingleton noNationSingleton = NoNationSingleton.getInstance();
            return noNationSingleton.getNation();
        }
    }
}
